packages/runtime/test-runtime-utils/src/mocks.ts

```diff
@@ -226,12 +224,24 @@ export class MockContainerRuntime {
 		this.clientSequenceNumber++;
 		switch (this.runtimeOptions.flushMode) {
 			case FlushMode.Immediate: {
+				const idAllocationOp = this.generateIdAllocationOp();
+				if (idAllocationOp !== undefined) {
+					this.submitInternal(idAllocationOp, clientSequenceNumber);
+				}
 				this.submitInternal(message, clientSequenceNumber);
 				break;
 			}
 
 			case FlushMode.TurnBased: {
-				this.outbox.push(message);
+				// Id allocation messages are directly submitted during the resubmit path
+				if (
+					(message.content as IMockContainerRuntimeSequencedIdAllocationMessage).contents
```

Comment:
looks like this is used a couple times. should refactor it out into a typeguard.

Reply:
good call. done.